app/controllers/concerns/shopify_app/ensure_installed.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module ShopifyApp
+  module EnsureInstalled
+    extend ActiveSupport::Concern
+    include ShopifyApp::RedirectForEmbedded
+
+    included do
+      if ancestors.include?(ShopifyApp::LoginProtection)
+        message = <<~EOS
+        We detected the use of incompatible concerns (EnsureInstalled and LoginProtection) in #{name},
+        which may lead to unpredictable behavior. In a future release of this library this will raise an error.
+        EOS
+
+        ShopifyApp::Logger.deprecated(message ,"22.0.0")
```

Comment:
If we want to do minimum runway of 90 days, I think we need to bump this to "23.0.0", since we do a major version update every ruby-api update.